### PR TITLE
perf: addition of liveness and readiness probes to mcad deployment

### PR DESCRIPTION
**Pull request by dimakis**
[linked here in MCAD](https://github.com/project-codeflare/multi-cluster-app-dispatcher/pull/414)

### CODE REVIEW

**Comment by astefanutti** [comment]
/hold

This requires an MCAD version upgrade.

File: config/internal/mcad/deployment.yaml.tmpl [hunk 1]
```diff
@@ -40,6 +40,20 @@ spec:
...
+              path: /healthz
+              port: 8081
+            initialDelaySeconds: 30
+            timeoutSeconds: 5
+            periodSeconds: 5
+          readinessProbe:
+            httpGet:
+              path: /readyz
+              port: 8081
+            initialDelaySeconds: 30
```
Comment (sutaakar) [inline config/internal/mcad/deployment.yaml.tmpl:54]: Isn't 30 seconds too much?
How fast does MCAD usually start?

Comment (sutaakar) [reply config/internal/mcad/deployment.yaml.tmpl:54]: I am thinking whether there is actually any need for `initialDelaySeconds` for MCAD.
IMHO this is mainly leveraged for slowly starting containers when there is no need to run readiness/liveness probes too soon.
The image pulling doesn't count to the startup time, it should start once pod is provisioned.

Comment (dimakis) [reply config/internal/mcad/deployment.yaml.tmpl:54]: so we're in agreement here? there is no need for the `initialDelaySeconds` ?

Comment (astefanutti) [reply config/internal/mcad/deployment.yaml.tmpl:54]: +1

**Pull request merged**